jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/assignment/impl/strategy/RoundRobinAssignmentStrategy.java

```diff
@@ -0,0 +1,130 @@
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.stream.Collectors;
+
+public class RoundRobinAssignmentStrategy implements AssignmentStrategy {
+    private static final Logger logger = LoggerFactory.getLogger(RoundRobinAssignmentStrategy.class);
+    private static final String IDENTIFIER = "RoundRobin";
+
+    private boolean active = IDENTIFIER.equals(System.getProperty("org.jbpm.task.assignment.strategy"));
+
+    private Map<String,CircularQueue> circularQueueMap = new ConcurrentHashMap();
+
+    private class CircularQueue<T> extends LinkedBlockingQueue<T> {
+        @Override
+        public synchronized T take() {
+            T headValue = null;
+            try {
+                headValue = super.take();
+                super.offer(headValue);
+            } catch (InterruptedException e) {
+                logger.error("Thread interrupted during the 'take' from a circular queue in the " +
+                                     "RoundRobinAssignmentStrategy",e);
+            }
+            return headValue;
+        }
+    }
+
+    @Override
+    public String getIdentifier() {
+        return IDENTIFIER;
+    }
+
+    @Override
+    public Assignment apply(Task task, TaskContext taskContext, String s) {
+        List<OrganizationalEntity> excluded = ((InternalPeopleAssignments)task.getPeopleAssignments()).getExcludedOwners();
+        UserInfo userInfo = (UserInfo) ((org.jbpm.services.task.commands.TaskContext)taskContext).get(EnvironmentName.TASK_USER_INFO);
+
+        // Get the the users from the task's the potential owners
+        List<OrganizationalEntity> potentialOwners = task.getPeopleAssignments().getPotentialOwners().stream()
+                .filter(oe -> oe instanceof User && !excluded.contains(oe))
+                .collect(Collectors.toList());
+
+        // Get the users belonging to groups that are potential owners
+        task.getPeopleAssignments().getPotentialOwners().stream().filter(oe -> oe instanceof Group)
+                .forEach(oe -> {
+                    Iterator<OrganizationalEntity> groupUsers = userInfo.getMembersForGroup((Group)oe);
+                    if (groupUsers != null) {
+                        groupUsers.forEachRemaining(user -> {
+                            if (user != null && !excluded.contains(user) && !potentialOwners.contains(user)) {
+                                potentialOwners.add(user);
+                            }
+                        });
+                    }
+                });
+        String queueName = getQueueName(task);
+        CircularQueue<OrganizationalEntity> mappedQueue = synchronizedQueue(circularQueueMap.get(queueName),potentialOwners);
+        // if the queue did not already exist then it will be added to the map
+        // otherwise the queue that was in the map will be updated
+        circularQueueMap.put(queueName,mappedQueue);
```

Comment:
I'd recommend to move this into the synchronizedQueue method to avoid potential race condition in case multiple threads would come in to get the queue as they might get new queues from synchronizedQUeue method and then replace each other on this line